extensions/azurePublishNew/src/components/shared/jsonEditor/JsonEditor.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/** @jsx jsx */
+import { jsx, css, SerializedStyles } from '@emotion/core';
+import React, { useRef } from 'react';
+import MonacoEditor, { EditorDidMount, EditorProps } from '@monaco-editor/react';
+import { NeutralColors, SharedColors } from '@uifabric/fluent-theme';
+import { MessageBar, MessageBarType, Link } from 'office-ui-fabric-react';
+import formatMessage from 'format-message';
+import styled from '@emotion/styled';
+import { useShellApi } from '@bfc/extension-client';
+
+const FullWidthMessageBar = styled(MessageBar)`
+  width: 100vw;
+`;
+
+const styles = {
+  container: ({ hovered, focused, error, width }) => {
+    let borderColor = NeutralColors.gray120;
+
+    if (hovered) {
+      borderColor = NeutralColors.gray160;
+    }
+
+    if (focused) {
+      borderColor = SharedColors.cyanBlue10;
+    }
+
+    if (error) {
+      borderColor = SharedColors.red20;
+    }
+    return css`
+      border-width: ${focused ? '2px' : '1px'};
+      padding: ${focused ? 0 : '1px'};
+      border-style: solid;
+      border-color: ${borderColor};
+      transition: border-color 0.1s linear;
+      box-sizing: border-box;
+      width: ${typeof width === 'string' ? width : `${width}px`};
+      label: BaseEditor;
+    `;
+  },
+};
+
+type Props = {
+  onChange?: (newValue: string) => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  styleOverrides?: SerializedStyles[];
+  errorMessage: string;
+} & Omit<EditorProps, 'options' | 'language' | 'editorDidMount'>;
+
+const editorOptions = {
+  scrollBeyondLastLine: false,
+  scrollbar: {
+    alwaysConsumeMouseWheel: false,
+  },
+  wordWrap: 'off' as any,
+  wordWrapColumn: 120,
+  fontFamily: 'Courier',
+  fontSize: '14',
+  fontWeight: '600',
+  lineNumbers: 'off' as any,
+  quickSuggestions: false,
+  minimap: {
+    enabled: false,
+    maxColumn: 0,
+  },
+  lineDecorationsWidth: 10,
+  lineNumbersMinChars: 3,
+  glyphMargin: false,
+  folding: false,
+  renderLineHighlight: 'none' as any,
+  formatOnType: true,
+  fixedOverflowWidgets: true,
+};
+
+const urls = {
```

Comment:
why is this need to be an object?

Reply:
I was trying to keep it consistent in all the components.